packages/presentation/lib/screen/login/login_screen.dart

```diff
@@ -60,92 +61,146 @@ class _LoginScreenState extends BlocScreenState<LoginScreen, LoginBloc> {
+                            width: Dimens.size0,
+                            style: BorderStyle.none,
+                          ),
                         ),
                       ),
                     ),
-                  ),
-                  const SizedBox(height: Dimens.size16),
-                  Text(
-                    AppLocalizations.of(context)!.password,
-                    style: AppTextStyles.sfProMediumUnselected12px,
-                  ),
-                  const SizedBox(height: Dimens.size8),
-                  LoginPasswordTextField(bloc: bloc),
-                  const SizedBox(height: Dimens.size32),
-                  SizedBox(
-                    width: width,
-                    height: Dimens.size48,
-                    child: ElevatedButton(
-                      onPressed: bloc.auth,
-                      style: ButtonStyle(
-                        backgroundColor: MaterialStateProperty.all(
-                          AppColorsDark.primaryColor,
-                        ),
+                    const SizedBox(height: Dimens.size16),
+                    Text(
+                      SM.current.password,
+                      style: AppTextStyles.sfProMediumUnselected12px,
+                    ),
+                    const SizedBox(height: Dimens.size8),
+                    TextFormField(
+                      controller: bloc.textPasswordController,
+                      obscureText: !_passwordVisible,
+                      autocorrect: false,
+                      enableSuggestions: false,
+                      cursorColor: Colors.white,
+                      style: const TextStyle(
+                        color: AppColorsDark.unselectedColor,
                       ),
-                      child: Text(
-                        AppLocalizations.of(context)!.login,
-                        style: AppTextStyles.sfProRegularUnselected16px,
+                      onChanged: (passwordText) {
+                        bloc.onChangedPassword(passwordText);
+                      },
+                      validator: (_) => bloc.validatePassword(),
+                      decoration: InputDecoration(
+                        prefixIcon: SvgPicture.asset(
+                          ImagesPath.lockImage,
+                          width: Dimens.size18,
+                          height: Dimens.size18,
+                          fit: BoxFit.none,
+                        ),
+                        suffixIcon: IconButton(
+                          icon: Icon(
+                            _passwordVisible
+                                ? Icons.visibility
+                                : Icons.visibility_off,
+                            color: Theme.of(context).primaryColorDark,
+                          ),
+                          onPressed: () {
+                            setState(() {
+                              _passwordVisible = !_passwordVisible;
+                            });
```

Comment:
hmm, i don't like, that you call setState in screen, because that should make block.
you have some variants
1)make method in bloc 
2)make widget AppTextField, which will contain TextFormField and there call setstate

Reply:
I choose make another widget